Mathlib/Combinatorics/Quiver/Path/Vertices.lean

```diff
@@ -160,7 +160,325 @@ lemma comp_eq_nil_iff {a b : V} {p : Path a b} {q : Path b a} :
 lemma end_mem_vertices {a b : V} (p : Path a b) : b ∈ p.vertices := by
   have h₁ : p.vertices.getLast (vertices_ne_nil p) = b :=
     vertices_getLast p (vertices_ne_nil p)
-  have h₂ := List.getLast_mem (l := p.vertices) (vertices_ne_nil p)
+  have h₂ := getLast_mem (l := p.vertices) (vertices_ne_nil p)
   simpa [h₁] using h₂
 
+/-!  ### Path vertices decomposition -/
+
+section
+
+variable {V : Type*} [Quiver V]
+
+open List
```

Comment:
You already have these `variable`s, and `List` is already open. OTOH I think (almost?) every lemma below consumes a path between two vertices so how about:
```suggestion
variable {a b : V} (p : Path a b)
```
and then you can drop these arguments from all the lemma statements.